File: php/src/CourtListener/Api/FinancialDisclosures.php
<?php

namespace CourtListener\Api;

use CourtListener\CourtListenerClient;
use CourtListener\Exceptions\CourtListenerException;

/**
 * FinancialDisclosures API endpoint
 */
class FinancialDisclosures extends BaseApi
{
    protected string $endpoint = 'financial-disclosures/';

    /**
     * Constructor
     *
     * @param CourtListenerClient $client
     */
    public function __construct(CourtListenerClient $client)
    {
        parent::__construct($client);
    }

    /**
     * List financial disclosures with advanced filtering
     *
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function listFinancialDisclosures(array $params = []) {
        return $this->list($params);
    }

    /**
     * Get a specific financial disclosures by ID
     *
     * @param int|string $id FinancialDisclosures ID
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function getFinancialDisclosures(array $params = []) {
        return $this->get($id, $params);
    }

    /**
     * Search financial disclosures
     *
     * @param array $params Search parameters
     * @return array
     * @throws CourtListenerException
     */
    public function searchFinancialDisclosures(array $params = []) {
        return $this->search($params);
    }
}

File: python/courtlistener/api/financial_disclosures.py
"""
Financial Disclosures API module for CourtListener SDK.
"""

from typing import Dict, Any, Optional, List
from .base import BaseAPI


class FinancialDisclosuresAPI(BaseAPI):
    """API for accessing financial disclosure reports."""
    
    def __init__(self, client):
        super().__init__(client)
        self.endpoint = "financial-disclosures/"
    
    def list(
        self,
        judge: Optional[int] = None,
        year: Optional[int] = None,
        **kwargs
    ) -> Dict[str, Any]:
        """
        List financial disclosure reports.
        
        Args:
            judge: Filter by judge ID
            year: Filter by disclosure year
            **kwargs: Additional query parameters
        
        Returns:
            API response with paginated list of financial disclosures
        """
        params = {}
        if judge is not None:
            params['judge'] = judge
        if year is not None:
            params['year'] = year
        
        params.update(kwargs)
        return self.client.get(self.endpoint, params=params)
    
    def get(self, disclosure_id: int) -> Dict[str, Any]:
        """
        Get a specific financial disclosure by ID.
        
        Args:
            disclosure_id: Financial disclosure ID
        
        Returns:
            Financial disclosure data
        """
        return self.client.get(f"{self.endpoint}{disclosure_id}/")
    
    def paginate(
        self,
        judge: Optional[int] = None,
        year: Optional[int] = None,
        **kwargs
    ):
        """
        Get paginated financial disclosures.
        
        Args:
            judge: Filter by judge ID
            year: Filter by disclosure year
            **kwargs: Additional query parameters
        
        Returns:
            PageIterator for financial disclosures
        """
        params = {}
        if judge is not None:
            params['judge'] = judge
        if year is not None:
            params['year'] = year
        
        params.update(kwargs)
        return self.client.paginate(self.endpoint, params=params)
